processor/src/main/java/org/mapstruct/ap/internal/util/Filters.java

```diff
@@ -111,7 +114,24 @@ public Map<String, ReadAccessor> recordAccessorsIn(Collection<Element> recordCom
     }
 
     private TypeMirror getReturnType(ExecutableElement executableElement) {
-        return getWithinContext( executableElement ).getReturnType();
+        TypeMirror returnType = getWithinContext( executableElement ).getReturnType();
+        if ( returnType.getKind() == TypeKind.WILDCARD ) {
+            // we got 2 versions of wildcards: '?' or '? extends ...'
+            if ( ( (WildcardType) returnType ).getExtendsBound() != null ) {
+                // in case of '? extends ...' we want to return the '...' part.
+                return ( (WildcardType) returnType ).getExtendsBound();
+            }
+            // in case of just '?', we want to look at the original methods return type and see if that one has an
```

Comment:
I guess this can only be if the method is generic. Because the `getWithinContext` would already have tackled the class defined type-var's right?

Reply:
That is correct.